playwright/tests/a11y.spec.js

```diff
@@ -1,66 +1,145 @@
+]
 
-test.describe('Accessibility Tests', async () => {
+test.describe(`Accessibility Site Scan`, async () => {
   test.setTimeout(4200000)
 
-  const pages = await getSitemapLocations(
-    process.env.BASE_URL || 'http://127.0.0.1:8001'
-  )
+  test(`The sitemap should contain accessible pages`, async ({
+    page,
+    browserName,
+    // makeAxeBuilder,
+  }, testInfo) => {
+    const viewportSize = page.viewportSize()
+    let scanResults = []
+    let failedPages = []
+    let pages = await getSitemapLocations(baseUrl)
 
-  let pageSegments
+    if (segmentNumber !== 0) {
+      pages = splitArray(pages, totalSegments)[segmentNumber - 1]
+    }
 
-  test.beforeAll(async () => {
-    pageSegments = await splitPagesIntoSegments(pages, 5)
-  })
+    console.log('number of pages in segment', pages.length)
+    console.log('browser name:', browserName)
+    console.log('viewport size:', viewportSize)
+
+    for (const pageUrl of pages) {
+      if (excludedPages.includes(pageUrl)) {
+        console.log('skipping scan of excluded page:', pageUrl)
+        continue
+      }
+
+      try {
+        console.log('testing page:', pageUrl)
+        await page.goto(pageUrl)
+
+        // @todo The shared "makeAxeBuilder" never reports errors for whatever reason so not using for now.
+        // const accessibilityScanResults = await makeAxeBuilder({ page }).analyze()
+
+        const accessibilityScanResults = await new AxeBuilder({ page })
+          .withTags(['section508', 'wcag2a', 'wcag2aa', 'wcag21a', 'wcag21aa'])
+          .exclude('iframe')
+          // Exclude header and footer since they are shared on every page.
+          .exclude('header')
+          .exclude('footer')
+          .analyze()
+
+        console.log('page violations:', accessibilityScanResults.violations)
+        if (accessibilityScanResults.violations.length > 0) {
+          scanResults.push({
+            url: pageUrl,
+            browserName,
+            viewportSize,
+            violations: accessibilityScanResults.violations,
+          })
+        }
+      } catch (error) {
+        let message = error
+        if (error instanceof Error) {
+          message = error.message
+        }
+
+        console.log('error:', message)
+        failedPages.push({
+          url: pageUrl,
+          browserName,
+          viewportSize,
+          error: message,
+        })
+      }
+    }
 
-  for (let i = 0; i < 5; i++) {
-    test(`the site should be accessible - Segment ${i + 1}`, async ({
-      page,
-      makeAxeBuilder,
-    }) => {
-      await runA11yTestsForPages(
-        pageSegments[i],
-        `Segment ${i + 1}`,
-        page,
-        makeAxeBuilder
+    if (scanResults.length > 0) {
+      // Remove root array from scanResults so we can merge results more cleanly.
+      const trimmedScanResults =
+        JSON.stringify(scanResults, null, 2).replace(/^\[|]$/g, '') +
```

Comment:
this is hard to parse at a glance. is it just I'm curious why we need to do this / add a trailing comma... Let's keep the individual scans as valid JSON imo.

something like
`jq -s '.' *.json | jq  'flatten' > merged-report.json` will take all json files and merge them into one array of objects without any fussing with removing brackets or adding commas to the individual files.

Reply:
Okay if that works, I'm fine with it.